modeshape-jcr/src/main/resources/org/modeshape/jcr/modeshape_builtins.cnd

```diff
@@ -140,6 +140,22 @@
     < 'STRING', 'URI', 'BINARY', 'LONG', 'DOUBLE', 'DECIMAL', 'BOOLEAN', 
       'DATE', 'NAME', 'PATH', 'REFERENCE', 'WEAKREFERENCE', 'UNDEFINED'
 
+// -----------------------
+// Large unordered collections
+// -----------------------
+// Abstract base type for all unordered collections.
+
+[mode:unorderedCollection] abstract
+  + * (nt:base) = nt:unstructured abort 
```

Comment:
At present, the `mode:unorderedCollection` is an abstract node type that allows any number of child nodes of any node type. Admittedly I designed it via this [MODE-2109 comment](https://issues.jboss.org/browse/MODE-2109?focusedCommentId=12928688&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-12928688). But doesn't this limit a user's ability to design their own structure? For example, if a user has a node type `acme:foo` that contains large numbers of `acme:bar` nodes, then the `acme:foo` must extend `mode:unorderedCollection` and thus would technically allow any numbers of non-`acme:bar` nodes, too.

Would it be better for `mode:unorderedCollection` to be mixins instead, with no restriction on the child node definitions other than they cannot allow SNS? The `NodeTypes` class could certainly enforce this and any other restrictions (like no versioning, etc.). And, I'm not sure much (if any) of the logic will need to change (other than any code that explicitly sets this), since a node's primary type or mixin types might ultimately extend `mode:unorderedCollection` the same way if it were a node type or a mixin type.

What do you think?